### PR TITLE
SharePoint primer under "Tools"

### PR DESCRIPTION
**Pull request by neilmb**
As 18F works more with partners who use Microsoft tools, SharePoint is becoming a much more common platform for 18F developers to be faced with. This document collects some notes from an engagement dealing with SharePoint and shares them here in the hope that they can be helpful to others in the future.

I'm not sure about placing this on the `tools` side navigation since it doesn't agree in scope with the other things in that nav menu, but I wasn't sure what a better place to link to it would be.

### CODE REVIEW

File: _guide/_pages/sharepoint.md [hunk 1]
```diff
@@ -0,0 +1,503 @@
+---
+title: A Modern SharePoint Primer
+sidenav: tools
+sticky_sidenav: true
+---
```
Comment (ryanhofdotgov) [inline _guide/_pages/sharepoint.md:5]:
```suggestion
subnav:
  - text: SharePoint Presentation Methods
    href: "#sharepoint-presentation-methods"
  - text: SharePoint Development Process
    href: "#sharepoint-development-process"
  - text: SharePoint and Microsoft Teams
    href: "#sharepoint-and-microsoft-teams"
  - text: PowerShell for SharePoint administration
    href: "#powershell-for-sharepoint-administration"
---
```

File: _guide/_data/navigation.yml [hunk 2]
```diff
@@ -45,6 +45,8 @@ tools:
     href: /continuous-deployment/
   - text: Datastore Selection
     href: /datastore-selection/
+  - text: Microsoft SharePoint
```
Comment (ryanhofdotgov) [inline _guide/_data/navigation.yml:48]:
```suggestion
  - text: SharePoint Primer
```

**Review by ryanhofdotgov — APPROVED** [review]
This is a great write-up – thanks! I made a couple of suggestions regarding the menu entry text and subnav entries; incorporate as you see fit.

You are right – the other articles under the Tools heading don't seem to match the flavor of this doc, but it seems the best place. I think over the next few months, we will start to see more product-specific pages appear under this section. There's a Docker page that could be moved to this level, and our backlog has some other tool-specific guidance that needs to be written up.

Out of curiosity, did you happen to evaluate any utilities that would allow you to track source code changes in git and publish to Sharepoint? The `Development` and `Deployment` stages of the workflow seem pretty painful.

**Comment by neilmb** [comment]
Unfortunately @ryanhofdotgov I didn't see anything that allows tracking changes to the site in Git. I can imagine that the PowerShell utilities could be used to build something like a static-site generator like Jekyll meant for Sharepoint, but I didn't find anything like that in existence.

**Pull request merged**